central/sensor/service/pipeline/nodeindex/pipeline.go

```diff
@@ -66,37 +71,36 @@ func (p pipelineImpl) Run(ctx context.Context, clusterID string, msg *central.Ms
 		log.Errorf("index report from node %s has unsupported action: %q", event.GetNode().GetName(), event.GetAction())
 		return nil
 	}
-	log.Debugf("received node index report with %d packages from %d content sets for node %s",
-		len(report.GetContents().Packages), len(report.GetContents().Repositories), event.GetId())
-	cr := report.CloneVT()
+	log.Debugf("received node index report for node %s with %d packages from %d content sets",
+		event.GetId(), len(report.GetContents().Packages), len(report.GetContents().Repositories))
+	report = report.CloneVT()
 
-	// Read the node from the database, if not found we fail.
-	node, found, err := p.nodeDatastore.GetNode(ctx, event.GetId())
+	// Query storage for the node this report comes from
+	nodeId := event.GetId()
+	node, found, err := p.nodeDatastore.GetNode(ctx, nodeId)
 	if err != nil {
-		return errors.WithMessagef(err, "fetching node: %s", event.GetId())
+		return errors.WithMessagef(err, "failed to fetch node %s from database", nodeId)
 	}
 	if !found {
-		return errors.WithMessagef(err, "node does not exist: %s", event.GetId())
+		return errors.WithMessagef(err, "node %s not found in datastore", nodeId)
 	}
 
-	// Send the Node and Index Report to Scanner for enrichment
-	err = p.enricher.EnrichNodeWithInventory(node, nil, cr)
+	// Send the Node and Index Report to Scanner for enrichment. The result will be persisted in node.NodeScan
+	err = p.enricher.EnrichNodeWithInventory(node, nil, report)
 	if err != nil {
-		return errors.WithMessagef(err, "enriching node %s with index report", event.GetId())
+		return errors.WithMessagef(err, "enriching node %s with index report", node.GetId())
 	}
-	log.Infof("Successfully enriched node %s with index report.", node.GetName())
+	log.Infof("Successfully enriched node %s with %s report - found %d components (id: %s)",
```

Comment:
Let's optimize this log line a bit. I understand that the results are already being stored in the DB (cause I see it in the UI), so maybe we should drop that line?
It looks a bit chatty in the logs - there are two lines saying almost the same:

```
central-586db8854-tpfnb pkg/scanners/scannerv4: 2024/10/15 08:31:07.853708 scannerv4.go:250: Info: Received Vulnerability Report with 542 packages containing 560 vulnerabilities
central-586db8854-tpfnb sensor/service/pipeline/nodeindex: 2024/10/15 08:31:07.859547 pipeline.go:93: Info: Successfully enriched node mm-10-14-encouraging-94ntm-worker-c-9h852 with SCANNER_V4 report - found 542 components (id: 60663bf0-58cd-4b49-8239-bf2bd2137a9d)
central-586db8854-tpfnb pkg/scanners/scannerv4: 2024/10/15 08:31:15.107541 scannerv4.go:250: Info: Received Vulnerability Report with 542 packages containing 560 vulnerabilities
central-586db8854-tpfnb sensor/service/pipeline/nodeindex: 2024/10/15 08:31:15.113582 pipeline.go:93: Info: Successfully enriched node mm-10-14-encouraging-94ntm-worker-b-htz6z with SCANNER_V4 report - found 542 components (id: 4b02661a-04fc-4c0d-bc2a-6445274a0742)
central-586db8854-tpfnb pkg/scanners/scannerv4: 2024/10/15 08:31:31.693558 scannerv4.go:250: Info: Received Vulnerability Report with 542 packages containing 560 vulnerabilities
central-586db8854-tpfnb sensor/service/pipeline/nodeindex: 2024/10/15 08:31:31.699185 pipeline.go:93: Info: Successfully enriched node mm-10-14-encouraging-94ntm-master-1 with SCANNER_V4 report - found 542 components (id: ad4859e3-f938-429d-9fee-58ddd2bdfdc3)
central-586db8854-tpfnb cloudsources/manager: 2024/10/15 08:31:55.329375 manager.go:176: Info: Fetching discovered clusters from cloud sources
central-586db8854-tpfnb cloudsources/manager: 2024/10/15 08:31:55.330646 manager.go:179: Info: Received 0 discovered clusters from cloud sources
central-586db8854-tpfnb pkg/scanners/scannerv4: 2024/10/15 08:32:01.878801 scannerv4.go:250: Info: Received Vulnerability Report with 542 packages containing 560 vulnerabilities
central-586db8854-tpfnb sensor/service/pipeline/nodeindex: 2024/10/15 08:32:01.886711 pipeline.go:93: Info: Successfully enriched node mm-10-14-encouraging-94ntm-master-2 with SCANNER_V4 report - found 542 components (id: f51f80ce-e91d-4165-a123-3650a7b85005)
```

Reply:
I've bumped it down to Debug logs for now, we don't need this in normal operation once this feature is completed